### PR TITLE
Sorting the job summaries while displaying

### PR DESCRIPTION
**Pull request by diptanu**

### CODE REVIEW

**Comment by dadgar** [comment]
Build an array of the task group names at line :259, sort.Strings(tgs), and then iterate over the sorted list and do a map lookup to get the summary per task group and then add it to the output in sorted order. No need to do the sort on the outputted data.

**Comment by diptanu** [comment]
This was done so that we don't iterate over the map twice, but since the sort interface is going to be cleaner and the output slice isn't too big we can go over the map twice.

File: command/status.go [hunk 1]
```diff
@@ -257,14 +258,20 @@ func (c *StatusCommand) outputJobInfo(client *api.Client, job *api.Job) error {
...
-			summaries[idx] = fmt.Sprintf("%s|%d|%d|%d|%d|%d|%d",
-				tg, tgs.Queued, tgs.Starting,
+		i := 0
+		taskGroups := make([]string, len(summary.Summary))
+		for taskGroup := range summary.Summary {
+			taskGroups[i] = taskGroup
+			i += 1
+		}
+		sort.Sort(TaskGroupSort(taskGroups))
+		for idx, taskGroup := range taskGroups {
```
Comment (dadgar) [inline command/status.go]:
```
		taskGroups := make([]string, 0, len(summary.Summary))
		for taskGroup := range summary.Summary {
			taskGroups = append(taskGroups, taskGroup)
		}
		sort.Strings(taskGroups)
```

File: command/status.go [hunk 2]
```diff
@@ -374,3 +381,19 @@ func convertApiJob(in *api.Job) (*structs.Job, error) {
 	}
 	return structJob, nil
 }
+
+// TaskGroupSort implements the sort interface and sorts an array of task group
+// names
+type TaskGroupSort []string
```
Comment (dadgar) [inline command/status.go]: Remove this and test

**Pull request merged**